standards/application/p2p-reliability.md

```diff
@@ -0,0 +1,328 @@
+Such clients SHOULD filter out duplicate messages by comparing [deterministic message hashes](https://rfc.vac.dev/waku/standards/core/14/message#deterministic-message-hashing).
+Note that both bandwidth usage and computational complexity increases proportionally to the amount of service nodes used.
+For this reason, we RECOMMEND using only two filter service nodes at a time.
+
+#### 3. Maintaining healthy subscriptions
+
+- A filter client using this strategy MUST regularly send a [`SUBSCRIBER_PING`](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscriber_ping) to each of its filter service nodes,
+to ensure that the service node is online and maintaining an active subscription for the client.
+The interval between each `SUBSCRIBER_PING` is up to the implementation,
+but we RECOMMEND `1 minute`.
+- If the `SUBSCRIBER_PING` request times out or returns an error code,
+the client SHOULD attempt to reinstall its filter subscription with a [`SUBSCRIBE`](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscribe) request.
+- If a [`SUBSCRIBER_PING`] or [`SUBSCRIBE`] request fails more than once to the same filter service node,
+the client MAY choose to replace this service node with another from the service node pool.
+We RECOMMEND a strict policy of replacing filter service nodes after only two `SUBSCRIBER_PING` failures
+or after a single `SUBSCRIBE` failure.
+- A filter client MAY also choose to refresh its existing subscriptions periodically,
+by submitting the same filter criteria as before in a new [`SUBSCRIBE`](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md#subscribe) request to the same service node.
+This helps ensure that local and remote views of filter criteria remains synchronised.
+
+#### 4. Store query on loss detection
+
+> *_Note:_* Filter clients participating in [Store-based reliability](#store-based-reliability) already performs this strategy and can ignore this section.
+
+- A filter client using this strategy MUST use [13/WAKU2-STORE](https://rfc.vac.dev/waku/standards/core/13/store) queries to retrieve lost messages.
+- Such clients MAY use [Store-based reliability](#store-based-reliability) to periodically detect and remedy message losses.
+- If [Store-based reliability](#store-based-reliability) is unsuitable (e.g. due to the high resource usage of repeated store queries),
+the client MAY perform opportunistic store queries covering periods over which it detected a disconnection.
+For example, a client MAY consider itself offline over a period of repeated failed regular pings and perform a store query once the connection has been restored.
+The store query SHOULD cover the period from the last successfully received message to the reestablishment of connectivity.
```

Comment:
Maybe an implementation detail. The current strategy to decide on the `start` time of a store time range query is using the timestamp of the last successful store query.

Reply:
Indeed. I don't think the spec precludes this as implementation, but simply suggests a time period that would be sufficient.